ethcore/src/miner/mod.rs

```diff
@@ -184,6 +184,14 @@ pub trait MinerService : Send + Sync {
 	fn ready_transactions<C>(&self, chain: &C, max_len: usize, ordering: PendingOrdering) -> Vec<Arc<VerifiedTransaction>>
 		where C: ChainInfo + Nonce + Sync;
 
+	/// Get a list of all ready transactions either ordered by priority or unordered (cheaper), optionally filtered by hash, sender or receiver.
+	///
+	/// Depending on the settings may look in transaction pool or only in pending block.
+	/// If you don't need a full set of transactions, you can add `max_len` and create only a limited set of
+	/// transactions.
+	fn ready_transactions_filtered<C>(&self, chain: &C, max_len: usize, tx_hash: Option<H256>, receiver: Option<H160>, sender: Option<H160>, ordering: PendingOrdering) -> Vec<Arc<VerifiedTransaction>>
```

Comment:
What about filtering by other stuff like gas price, value, gas, etc? I guess it might be more complicated, since you will most likely want greater than/lesser than operators.

I think I'd rather be for a `FilteringObject` instead of passing all parameters by index. So:

```rust
/// This will require a custom serde impl, to handle cases like:
/// { "gt": "0x..." }, { "lt": "0x..." }, "0x..."
/// The last case should resolve to `Equal` case.
enum Operator<T> {
  /// Don't filter by this field
  Any,
  /// Require the value to be equal
  Equal(T),
  /// Require the value to be greater than
  GreaterThan(T),
  /// Require the value to be less than
  LessThan(T),
}
#[serde(Serialize, Deserialize)]
struct ReadyTransactionsFilter {
   sender: Operator<Address>,
   receiver: Operator<Option<Address>>,
   gas: Operator<U256>,
   gas_price: Operator<U256>,
   value: Operator<U256>,
   // I don't see use cases for this, but we might add it for completeness.
   nonce: Operator<U256>,
   /// We can also unify this limitting parameter to be passed here,
   /// but it's not strictly required since it has different semantics, so your call on this one :)
   // max_results: usize,
}

/// then you could unify the methods:
fn ready_transactions(&self, chain, max_len, filter: Option<ReadyTransactionsFilter>)

Reply:
> What about filtering by other stuff like gas price, value, gas, etc? I guess it might be more complicated, since you will most likely want greater than/lesser than operators.

It sure looks more complicated :) But I can come up with something. Would you like to have all those changes included in one round or should I first complete the rest and follow up with the gas/price filtering implementation later on? I'm also curious about performance on this one, but let's see.